src/hooks/usePublicResolverAddress.ts

```diff
@@ -0,0 +1,22 @@
+import { useEns } from '@app/utils/EnsProvider'
+import { useQuery } from 'react-query'
+
+export const usePublicResolverAddress = () => {
```

Comment:
you can make this synchronous if you use the getContractAddress function (just needs chainid), should probably add a generic useContractAddress hook but i'll do that separately

Reply:
useContractAddress:

```ts
import { useEns } from '@app/utils/EnsProvider'
import { ContractName } from '@ensdomains/ensjs/dist/cjs/contracts/types'
import { useChainId } from './useChainId'

export const useContractAddress = (contractName: ContractName) => {
  const chainId = useChainId()
  const { getContractAddress } = useEns()
  return getContractAddress(chainId as any)(contractName)
}
```

Reply:
added